File: 프로그래머스/2/389479. 서버 증설 횟수/서버 증설 횟수.java
class Solution {
    public int solution(int[] players, int m, int k) {
        int answer = 0;
        int[] server_count = new int[24];

        for(int i = 0 ; i < 24; i ++){
            int need = players[i] / m;
            int more = need - server_count[i];
            if(more > 0){
                answer += more;
                for(int j = i ; j < Math.min(i + k, 24) ; j++){
                    server_count[j] += more;
                }
            }
        }
        return answer;
    }
}

File: 프로그래머스/2/389479. 서버 증설 횟수/서버 증설 횟수.kt
import kotlin.math.min
import kotlin.math.max
class Solution {
    fun solution(players: IntArray, m: Int, k: Int): Int {
        var answer: Int = 0
        val tracking = IntRange(0, 24).map { 0 }.toMutableList()

        for(i in IntRange(0, 23)){
            var up = players[i] / m
            up = max(0, up - tracking[i])
            if(up > 0){

                for(j in i until min(24, i+k)){
                    tracking[j] += up
                }
    //            println("i: $i, up: $up")
                answer += up
            }
        }
        return answer
    }
}